Mathlib/Logic/Embedding/Set.lean

```diff
@@ -147,3 +152,61 @@ theorem subtypeOrEquiv_symm_inr (p q : α → Prop) [DecidablePred p] (h : Disjo
 #align subtype_or_equiv_symm_inr subtypeOrEquiv_symm_inr
 
 end Subtype
+
+section Disjoint
+
+variable {α ι : Type*} {s t r : Set α}
+
+/-- For disjoint `s t : Set α`, the natural injection from `↑s ⊕ ↑t` to `α`. -/
+@[simps] def Function.Embedding.sumSet (h : Disjoint s t) : s ⊕ t ↪ α where
+  toFun := Sum.elim (↑) (↑)
+  inj' := by
+    rintro (⟨a, ha⟩ | ⟨a, ha⟩) (⟨b, hb⟩ | ⟨b, hb⟩)
+    · simp [Subtype.val_inj]
+    · simpa using h.ne_of_mem ha hb
+    · simpa using h.symm.ne_of_mem ha hb
+    simp [Subtype.val_inj]
+
+@[norm_cast] lemma Function.Embedding.coe_sumSet (h : Disjoint s t) :
+    (Function.Embedding.sumSet h : s ⊕ t → α) = Sum.elim (↑) (↑) := rfl
+
+@[simp] theorem Disjoint.sumSubtypeEmbedding_preimage_inl (h : Disjoint s t) :
+    .inl ⁻¹' (Function.Embedding.sumSet h ⁻¹' r) = r ∩ s := by
+  ext
+  simp
+
+@[simp] theorem Disjoint.sumSubtypeEmbedding_preimage_inr {s t r : Set α} (h : Disjoint s t) :
+    .inr ⁻¹' (Function.Embedding.sumSet h ⁻¹' r) = r ∩ t := by
```

Comment:
Those brackets everywhere are unnecessary
```suggestion
    .inr ⁻¹' Function.Embedding.sumSet h ⁻¹' r = r ∩ t := by
```

Reply:
Doesn't work.

Reply:
Uh, we should fix that. Pretty sure it worked in Lean 4. But okay for now